testfixtures/comparison.py

```diff
@@ -685,3 +685,27 @@ def diff(x, y, x_label='', y_label=''):
             y_label or 'second',
             lineterm='')
     )
+
+
+class RangeComparison:
+    """
+    An object that can be used in comparisons of numerics to validate
+    actual falls within a certain range
+    """
+    def __init__(self, lower_bound, upper_bound):
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+    def __eq__(self, other):
+        try:
+            0 + other
```

Comment:
@asqui I was playing around with putting the try catch around the actual operator as you suggested but it seems like python hardly throws a type error hardly for any object when using the `<=` or `>=` and just seems to return False in all the cases I tested.  I suppose that will be sufficient but I'm a bit on the fence if I prefer the user to see a TypeError or just get the normal assertion error when they compare the object.

What's your opinion?